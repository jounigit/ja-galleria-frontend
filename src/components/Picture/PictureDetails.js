import React, { useContext } from 'react'
import { Card, Image } from 'semantic-ui-react'
import { AuthContext } from '../../App'
import RemovePicture from './RemovePicture'

const PictureDetails = ({ picture }) => {
  const { state } = useContext(AuthContext)

  return (
    <div data-cy='picture'>
      <Card>
        <Image src={picture.thumb} wrapped ui={false} />
        <Card.Content header={picture.title} />
        <Card.Content>
          <Card.Meta>
            <span className='date'>{picture.created_at}</span>
          </Card.Meta>
        </Card.Content>
        <Card.Content description={picture.content} />
        <Card.Content extra>
          { state.user &&
              <RemovePicture
                id={ picture.id }
                title={ picture.title }
              />
          }
        </Card.Content>
      </Card>
    </div>
  )
}

export default PictureDetails