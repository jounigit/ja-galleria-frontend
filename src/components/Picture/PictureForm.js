import React from 'react'
import { Header, Image, Form, Input, Button, Container, TextArea } from 'semantic-ui-react'

const PictureForm = ({
  errorMessage,
  title,
  content,
  thumb,
  handleFormSubmit,
  // handleFileInputChange,
  handleInputChange
}) => {
  console.log('form ---', thumb)
  return (
    <Container>
      {errorMessage && (
        <Header as='h4' color='red' data-cy='error-message'>{errorMessage}</Header>
      )}


      <Form>
        <Image src={thumb} wrapped ui={false} />
        {/* <Form.Field>
          <label>
            Image
            <Input
              data-cy='file'
              type='file'
              value={file}
              onChange={handleFileInputChange}
              name='file'
            />
          </label>
        </Form.Field> */}
        <Form.Field>
          <label>
            title
            <Input
              data-cy='title'
              type='title'
              value={title}
              onChange={handleInputChange}
              name='title'
              id='title'
            />
          </label>
        </Form.Field>
        <Form.Field>
          <label>
            Content
            <TextArea
              data-cy='content'
              value={content}
              onChange={handleInputChange}
              name='content'
            />
          </label>
        </Form.Field>

        {/* <input type="button" value="Upload as Form" onClick={handleFormSubmit} /> */}

        <Button data-cy='submit' type='submit'  value="Upload" onClick={handleFormSubmit} />
      </Form>
    </Container>
  )
}

export default PictureForm