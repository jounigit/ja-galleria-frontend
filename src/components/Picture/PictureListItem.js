import React from 'react'
import { Image, Container, Popup } from 'semantic-ui-react'

const PictureListItem = ({ picture }) => {

  // console.log('DETAIL ---', picture)
  return (
    <div data-cy='pictureListItem'>
      <Container>

        <Popup
          content={picture.content}
          header={picture.title}
          trigger={<Image src={picture.image} size='medium' fluid />}
        />

      </Container>

    </div>
  )
}

export default PictureListItem

{/* <Image src={picture.image} size='medium' fluid />
      <Header as='h2'>{picture.title}</Header> */}
{/* <Segment> */}
{/* <Image src={picture.image} size='medium' fluid /> */}
{/* <Image src={picture.image} size='medium' centered /> */}
{/* <Header as='h2'>{picture.title}</Header> */}
{/* {picture.content} */}
{/* </Segment> */}

{/* <Card>
          <Image src={picture.thumb} wrapped ui={false} />
          <Card.Content header={picture.title} />
          <Card.Content>
            <Card.Meta>
              <span className='date'>{picture.created_at}</span>
            </Card.Meta>
          </Card.Content>
          <Card.Content description={picture.content} />
        </Card> */}