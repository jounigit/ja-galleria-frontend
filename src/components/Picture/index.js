export { default as PictureList } from './PictureList'
export { default as Picture } from './Picture'
export { default as PictureData } from './PictureData'