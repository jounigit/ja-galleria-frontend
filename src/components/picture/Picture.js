import React from 'react'

const Picture = ({picture}) => {
    return (
        <div className='picture'>
            <img src={picture.thumb} className="App-logo" alt={picture.thumb} />
            <h2>
                {picture.title}
            </h2>
            <p>
            {picture.content}
            </p>
        </div>
    )
}

export default Picture