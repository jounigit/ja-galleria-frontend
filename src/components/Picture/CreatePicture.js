import React, { useState, useContext } from 'react'
import {
//   Header,
//   Button,
  Container,
//   Modal,
//   Icon
} from 'semantic-ui-react'
import { PictureContext } from '../../contexts/PictureContext'
import apiService from '../../services/apiService'
import { CREATE_PICTURE } from '../../reducers/actionTypes'
import PictureForm from './PictureForm'

const initialState = {
  title: '',
  content: '',
  file: null,
  isSubmitting: false,
  errorMessage: null,
  message: null
}

const CreatePicture = () => {
  const [data, setData] = useState(initialState)
  //   const [formVisibility, setFormVisibility] = useState(false)
  const { dispatch } = useContext(PictureContext)

  // :::::::::::::::::::::::::::::::::::: //
  // hande input values
  const handleFileInputChange = event => {
    setData({
      ...data,
      file: event.target.files[0]
    })
  }
  const handleInputChange = event => {
    setData({
      ...data,
      [event.target.name]: event.target.value
    })
  }
  // handle errors
  const handleError = error => {
    setData({
      ...data,
      isSubmitting: false,
      errorMessage: error
    })
  }

  // console.log('PIC inputs ---', data)
  // ----- handle form submit - post new data ---------- //
  const handleFormSubmit = async(event) => {
    event.preventDefault()
    if(data.title === '') {
      return handleError('title is required!')
    }

    // const newData = {
    //   title: data.title,
    //   content: data.content,
    //   file: data.file
    // }
    const formData = new FormData()
    formData.append('image',data.file)
    formData.append('title',data.title)
    formData.append('content',data.content)

    try {
      const result = await apiService.create('pictures', formData)
      const newPicture = result.data
      console.log('NewPic ---', result)
      console.log('NewPic data ---', newPicture)

      dispatch({
        type: CREATE_PICTURE,
        data: newPicture
      })
      setData({
        title: '',
        content: '',
        file: null,
        isSubmitting: false,
        errorMessage: null,
        message: result.message
      })

    } catch (error) {
      console.error()
      handleError('failed storing picture!')
    }
  }

  return ( // <Modal as={Form} onSubmit={e => handleSubmit(e)} open={true} size="tiny">
    <Container>

      <PictureForm
        errorMessage={data.errorMessage}
        title={data.title}
        content={data.content}
        category_id={data.category_id}
        handleFormSubmit={handleFormSubmit}
        handleInputChange={handleInputChange}
        handleFileInputChange={handleFileInputChange}
        formHeader={'Uusi kuva'}
      />

    </Container>

  )
}

export default CreatePicture