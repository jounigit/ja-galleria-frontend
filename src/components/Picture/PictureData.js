import React, { Fragment }  from 'react'
import useFetch from '../../utils/useFetch'
import { useParams } from 'react-router-dom'
import Picture from './Picture'
const baseUrl = 'http://localhost:8000/api'

const PictureData = () => {
  let { id } = useParams()

  const picture = useFetch(
    `${baseUrl}/pictures/${id}`
  )

  if(picture.loading) {
    return <div className='loader'>Loading ...</div>
  }

  console.log('PICTUREDATA -', picture)

  return (
    <Fragment>
      {picture.data &&  <Picture picture={picture.data.data.data} />}
    </Fragment>
  )

}

export default PictureData