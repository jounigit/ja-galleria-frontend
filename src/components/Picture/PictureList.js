import React, { useState, useEffect } from 'react'
import pictureService from '../../services/pictureService'
import Picture from './Picture'

const PictureList = () => {
  const [Pictures, setPictures] = useState([])

  useEffect(() => {
    pictureService
      .getAll()
      .then(response => {
        setPictures(response.data)
      })
  }, [])

  const showPictures = () => Pictures.map(picture =>
    <Picture key={picture.id} picture={picture} />
  )

  return (
    <div>
      <h2>Kuvat</h2>
      { showPictures() }
    </div>
  )
}

export default PictureList