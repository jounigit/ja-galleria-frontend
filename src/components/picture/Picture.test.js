import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Picture from './Picture'


const picture = {
    title: 'Kuva 1',
    content: 'Hieno kuva',
    image: 'kuva1.jpg',
    thumb: 'kuva1.jpg'
}

describe('<Picture />', () => {
    let component

    beforeEach(() => {
        component = render(
            <Picture picture={picture} />
        )
    })

    test('picture img', () => {
        const img = component.container.querySelector('img')
        expect(img.alt).toEqual(picture.thumb)
    })

    test('picture title', () => {
        const h2 = component.container.querySelector('h2')
        expect(h2).toHaveTextContent(picture.title)
    })

    test('picture content', () => {
      expect(component.container).toHaveTextContent(
        'Hieno kuva'
      )
    })

    test('Picture renders correctly', () => {
        const tree = renderer
          .create(<Picture picture={picture} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      })

})

