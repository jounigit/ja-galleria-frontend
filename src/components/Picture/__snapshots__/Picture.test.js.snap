// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picture /> Picture renders correctly 1`] = `
<div
  className="picture"
>
  <img
    alt="kuva1.jpg"
    className="App-logo"
    src="kuva1.jpg"
  />
  <h2>
    Kuva 1
  </h2>
  <p>
    Hieno kuva
  </p>
</div>
`;
